### PR TITLE
Fix handle research in code generation

### PR DESCRIPTION
**Pull request by marius-pelegrin-arm**
The method `BaseGenerator.check_struct_pnext_handles` in `base_generator.py` should return `True` if the specified struct contains handles, handle pointers, or if a struct that contains handle or handle pointers can be referenced through it (either by the `pNext` chain or with a member that would be a struct or struct pointer).

Until now, to determine if a struct can reference such another struct, the method was retrieving `valid_extension_structs` which query structs that contains a field `structextends` with the checked struct. The issue is that if struct `A` and struct `B` both extend struct `C`, then struct `A` can be in the `pNext` chain of struct `B` (and otherwise) which was not detected.

This commit proposes to solve this issue by including in the `valid_extensions_structs` all the structs extending the parent struct of the checked struct.

### CODE REVIEW

**Comment by MarkY-LunarG** [comment]
@marius-pelegrin-arm, the codegen has changed drastically since you've made this change.  Can you verify if the current code suffers from the same issue?

**Comment by marius-pelegrin-arm** [comment]
The current code suffers from the same issue, that's certain, but I couldn't really rebase my change because I am not sure to understand the current code. You have at the same time some "global" dicts (`KhronosBaseGenerator.global_structs_with_handles`) and "local" dicts generated from `BaseGenerator.check_struct_member_handles` but the name is ambiguous as to what "with handles" means.

More precisely, the new "global" variable seems to not include AT ALL what happens in the pNext chains... while the `check_struct_member_handles` was handling pNext chain but just "badly" which is what the commit was trying to fix.

TLDR: The problem is still there but I am not sure what the code is trying to do so it's hard to find the proper way to fix it

**Comment by MarkY-LunarG** [comment]
@marius-pelegrin-arm , do you have a way of us testing this? 

Regarding the naming:  `check_struct_member_handles` should check to see that any structure member (or if it is another structure or pointer to possible structures like a `pNext`) may contain handles.  This is important to know in GFXR because during encode, we have to properly encode the correct handle information in place of those handles, and then during decode we need to replace the stored values with the correct ones at replay. 

The work has been tweaked yet again mostly due to findings while porting OpenXR.  But if you pointed us to an example (perhaps in private email) we could look at it and resolve the issue.
